onnx/version_converter/BaseConverter.h

```diff
@@ -0,0 +1,78 @@
+// Version converter interface for ONNX models between different opset versions.
+
+#pragma once
+
+#include "onnx/common/ir.h"
+#include "onnx/common/ir_pb_converter.h"
+#include "onnx/common/stl_backports.h"
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+#include <utility>
+#include <iostream>
+#include <stdlib.h>
+// TODO: Remove this import once actual adapters are imported
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+// TODO: Consider creating interface for this class.
+class BaseVersionConverter {
+  // Schema for adapters: {<op_name>:{<from_domain>$<from_version>:{<to_domain>
+  // <to_version>: adapter}}}
+  protected:
+    std::unordered_map<std::string, std::unordered_map<std::string, std::unordered_map<std::string, std::unique_ptr<Adapter>>>> adapters;
```

Comment:
Why do we still have the three-level map? Didn't we say a table (i.e. two-level) is enough?

Reply:
I think we decided that we wanted to maintain 3-level for the cross-domain case, since it made sense to maintain BaseConverter for that purpose.  That may have changed due to the abstraction of adapter_lookup to BaseConverter and the removal of IntraDomainConverter, though.  What do you think, @houseroad?

Reply:
2 level domain only works for intra/default domain case. The map and its lookup won't be a performance bottleneck. I would suggest to stick to this design, since later, we may have inter-domain converter.